src/dependency-manager/src/task-config/v1.ts

```diff
@@ -1,30 +1,136 @@
 import { Type } from 'class-transformer';
-import { Allow, IsEmpty, IsInstance, IsOptional, IsString } from 'class-validator';
-import { ServiceConfigV1 } from '../service-config/v1';
+import { Allow, IsEmpty, IsInstance, IsObject, IsOptional, IsString, Matches, ValidatorOptions } from 'class-validator';
+import { parse as shell_parse } from 'shell-quote';
+import { ResourceSpec, VolumeSpec } from '../common/base';
+import { BuildSpecV1, DeploySpecV1, ServiceVolumeV1, transformVolumes } from '../common/v1';
+import { Dictionary } from '../utils/dictionary';
+import { validateDictionary, validateNested } from '../utils/validation';
 import { TaskConfig } from './base';
 
 //TODO:84:Tasks: let's rethink our abstraction patterns a bit here. There's a chance we don't want TaskConfig extending ServiceConfig
-export class TaskConfigV1 extends ServiceConfigV1 implements TaskConfig {
+export class TaskConfigV1 extends TaskConfig implements ResourceSpec {
   @Allow({ always: true })
   __version?: string;
 
-  // TODO:84:Tasks: are there any cron parsers that we can use to validate?
   @IsOptional({ always: true })
   @IsString({ always: true })
-  schedule?: string;
+  @Matches(/^[a-zA-Z0-9-_]+$/, {
+    message: 'Names must only include letters, numbers, dashes, and underscores',
+  })
+  name?: string;
+
+  @IsOptional({ always: true })
+  @IsString({ always: true })
+  description?: string;
```

Comment:
Would prefer to not have all of this duplication of validation annotations. Can we have a ResourceSpecV1?

Reply:
**TLDR;** I'm open to alternatives, but I explored your proposed one and it leads to multiple inheritance.

The problem with the above is we have competing design concerns here between using inheritance for versioning (`*Spec extends *V1`) and using inheritance to re-use properties/annotations (`Task/Service* extends Resource*`):

ie:
```
// to re-use properties/annotations:
ResourceConfig extends BaseConfig
ServiceConfig extends ResourceConfig
TaskConfig extends ResourceConfig

// to version:
ResourceConfig extends BaseConfig
ResourceConfigV1 extends ResourceConfig

// which means we're stuck with:
TaskConfigV1 extends ?
ServiceConfigV1 extends ?
```
We might want TaskConfigV1 to extend both ResourceV1 _and_ TaskConfig but Typescript doesn't support multiple inheritance (I think for good reason... this is problematic):

```
TaskConfigV1 extends TaskConfig, ResourceConfigV1
ServiceConfigV1 extends ServiceConfig, ResourceConfigV1
```

So my solution was to use an interface to at least track the shared properties, and stick with using inheritance for versioning:
```
// use inheritence for versioning:
ServiceConfigV1 extends ServiceConfig
TaskConfigV1 extends TaskConfig

// use interface for tracking shared properties
interface ResourceConfig

ServiceConfig implements ResourceConfig
TaskConfig implements ResourceConfig
```

Reply:
Can we do?
`ServiceConfigV1 extends ResourceConfigV1  implements ServiceConfig`
`TaskConfigV1 extends ResourceConfigV1  implements TaskConfig`